06-reactJs/components/input.js

```diff
@@ -0,0 +1,36 @@
+var React = require('react');
+var _ = require('lodash');
+
+var Input = React.createClass({
+    
+    propTypes: {
+            defaultValue: React.PropTypes.string,
+            onChange: React.PropTypes.func,
+            value: React.PropTypes.string,
+            type: React.PropTypes.string
+    },
+
+    getInitialState: function () {
+        return {value=''}
+    },
+
+    render: function () {
+        return (
+            <input {...this.getProps} />
+        );
+    },
+
+    getProps: function () {
+        return {
+            type="text",
+            value= this.props.value,
+            onChange= this.handleChange()
```

Comment:
Careful here! We're passing a method as a callback, not executing it (parenthesis shouldn't be there).